src/Components/WebAssembly/WebAssembly.Authentication/src/Services/ExpiredTokenException.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// An <see cref="Exception"/> that is thrown when an <see cref="AuthorizationMessageHandler"/> instance
+    /// is not able to provision an access token.
+    /// </summary>
+    public class AccessTokenNotAvailableException : Exception
+    {
+        private readonly NavigationManager _navigation;
+        private readonly AccessTokenResult _tokenResult;
+
+        public AccessTokenNotAvailableException(
+            NavigationManager navigation,
+            AccessTokenResult tokenResult,
+            IEnumerable<string> scopes)
+            : base(message: "Unable to provision an access token for the requested scopes: " +
```

Comment:
You could override the `Message` property instead? Might be easier to write this.

Reply:
That sounds more trouble than what it's worth, IMO, as it forces me to capture and store the scopes on the exception. We can change in the future if we need/have to.